### DISCUSSION TITLE
Make it easier to figure out corruptions

### DISCUSSION THREAD

**Discussion by vivek-rane** [foxglove/mcap#1097] [General]
Right now if I run `mcap merge` on a large number of files I get an error message `failed to pull next message: unexpected literal count, want 48 bytes, but only 38 is available` but it doesn't tell me which file it is.

`mcap doctor` doesn't accept wildcards so I'll have to run it on each of the 500 files individually (or via script).

There needs to be an easier way to find which file is corrupt.

**Comment by jtbandes** [comment]
Internal tracking issue: https://linear.app/foxglove/issue/FG-2888

**Comment by james-rms** [comment]
> mcap doctor doesn't accept wildcards

Pedantic note: neither does `mcap merge`, it just lets the shell expand the wildcard into a sequence of filenames. The key difference is that `mcap merge` takes multiple file arguments, and `mcap doctor` only takes one. We could definitely add the ability for `mcap doctor` to take multiple file arguments.
